### ISSUE TITLE
Roles and Security

### ISSUE THREAD

**Issue by fragbait** [msgphp/msgphp#235]
While I have been looking at the demo and reading the notes, I have the current issue.

I have successfully created a role (ROLE_SUPERADMIN)
I have successfully assigned that role to a user

In security.yaml I have created the following access_control
         - { path: ^/admin, roles: ROLE_SUPERADMIN }

When I go to ./admin I get access denied.
Put it back to 
- { path: ^/admin, roles: IS_AUTHENTICATED_FULLY }

and it all works.
I admit I am following FOS user rule of thumb, and am unsure what I have messed up.

Thanks

**Comment by ro0NL** [comment]
Hi @fragbait 

Can you confirm you've configured role providers in v0.7?

See https://github.com/msgphp/symfony-demo-app/pull/65/files#diff-beb14464e85f87eae15acc81087abeb9R28

**Comment by MusterOhneWert** [comment]
Hello, I ran into the same issue. I'm very new at Symfony, so I can't really figure out, whats about "configured role providers" . In security.yaml, I found this entry:

`  providers:
        in_memory: { memory: ~ }
        msgphp_user: { id: MsgPhp\User\Infra\Security\SecurityUserProvider }`

This was inserted automatically into file, I didn't change it. This even looks for me, everything should work as meant. Or is there something missing? There are other entries in security.yaml with msgphp:

`main:
            anonymous: true
            provider: msgphp_user`

`encoders:
        MsgPhp\User\Infra\Security\SecurityUser: bcrypt`

`form_login:
                login_path: /login
                check_path: /login
                default_target_path: /profile
                username_parameter: nickname
                password_parameter: password

            logout:
                path: /logout
                target: /login
                invalidate_session: true`

**Comment by ro0NL** [comment]
You should look for the msgphp user-bundle configuration, i.e. `config/packages/msgphp_user.yaml`.

Then configure e.g.

```yaml
msgphp_user: 
  # ...
  role_providers: 
    - 
      default: 
        - ROLE_USER
    - MsgPhp\User\Role\UserRoleProvider
```

**Comment by MusterOhneWert** [comment]
Found second in 'config/packages/masgphp_user.make.php'

'return function (ContainerConfigurator $container) {
    $container->extension('msgphp_user', array (
  'class_mapping' => 
  array (
    'MsgPhp\\User\\Entity\\Role' => 'App\\Entity\\User\\Role',
    'MsgPhp\\User\\Entity\\UserRole' => 'App\\Entity\\User\\UserRole',
  ),
  'role_providers' => 
  array (
    'default' => 
    array (
      0 => 'USER',
    ),
    0 => 'MsgPhp\\User\\Role\\UserRoleProvider',
  ),
));'

**Comment by ro0NL** [comment]
Ah great, the 2nd is generated by `make:user:msgphp` and cab be merged into one file (but it's not required to do so), as Symfony will load all these files anyway.

based on that, you have role_providers already configured :)

From the original issue.. did you configured "access_control" in security.yaml? Or what's the bug on your side then?

**Comment by MusterOhneWert** [comment]
Thank you for quick help. I'll try to extract the necessary parts, for I'm not allowed to share code at all.

**Comment by MusterOhneWert** [comment]
Ohhhhh, I'm really really stupid. I tried to reproduce the steps an getting all parts needed - the error didn't appear again. Huh? Now, I think I get the point. The User was logged in while I changed his roles. After logout and login again, everything's fine. It seems the userroles weren't updated immediatly - but I only checked if the role changes affects the user by only reloading the page. If this is correct an should behave that way, so it is my own fault - really sorry for it!
